src/fdw/csv.rs

```diff
@@ -110,20 +110,11 @@ impl ForeignDataWrapper<BaseFdwError> for CsvFdw {
                 FOREIGN_DATA_WRAPPER_RELATION_ID => {}
                 FOREIGN_SERVER_RELATION_ID => {}
                 FOREIGN_TABLE_RELATION_ID => {
-                    let valid_options: Vec<String> = CsvOption::iter()
-                        .map(|opt| opt.as_str().to_string())
-                        .collect();
-
-                    validate_options(opt_list.clone(), valid_options)?;
-
-                    for opt in CsvOption::iter() {
-                        if opt.is_required() {
-                            check_options_contain(&opt_list, opt.as_str())?;
-                        }
-                    }
+                    validate_mapping_option::<CsvOption>(opt_list)?;
+                }
+                _ => {
```

Comment:
This feels unsafe. I'm not sure if we want to match all other cases to user mapping relation id. I made a PR to pgrx a while back to expose `USER_MAPPING_ID` to `pg_sys` but it would require we upgrade `pgrx`.

Reply:
I believe the latest version of `pgrx` has exposed `USER_MAPPING_ID`, but it's not a beta release. Perhaps we should consider upgrading `pgrx` to the latest stable version later.

Reply:
The next version is coming within the week! Should we wait for v0.12.0 before getting this in?

Reply:
Yes. It is better to wait for #66 to be resolved.

Reply:
Sweet, let's do that then. I'll convert this to a draft. Thank you for your help @Weijun-H!

Reply:
We've upgraded to pgrx 0.12.1, so we should be good!